diff --git i/mypy/main.py w/mypy/main.py
index 04442ad..682ec52 100644
--- i/mypy/main.py
+++ w/mypy/main.py
@@ -4,4 +4,4 @@ import sys
 from mypy.main import main
 
 if __name__ == '__main__':
-    main(None, sys.stdout, sys.stderr)
+    main()

diff --git i/mypy/BUILD w/mypy/BUILD
index 63bb405..0d3c3ef 100644
--- i/mypy/BUILD
+++ w/mypy/BUILD
@@ -1,4 +1,5 @@
 load("@mypy_integration_pip_deps//:requirements.bzl", "requirement")
+load("@ncore_pip_deps//:requirements.bzl", local_req = "requirement")
 
 py_binary(
     name = "mypy",
@@ -8,5 +9,8 @@ py_binary(
     visibility = ["//visibility:public"],
     deps = [
         requirement("mypy"),
+        # List of dependencies that provide type-hints
+        local_req("click"), local_req("pytest"), local_req("numpy"), local_req("sphinx"), local_req("multimethod"), local_req("jupyter_server"), local_req("dataclasses_json"), local_req("torch"),
+        "@ncore_repo//ncore/data:pylib", "@ncore_repo//ncore/data/v3:pylib", "@ncore_repo//ncore/geometry:pylib", "@ncore_repo//ncore/sensors:pylib",
     ],
 )
